pyfilesync/console.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Program makes backup of your file
+Example usage
+=============
+
+To invoke program type in your console::
+
+    filesync --help
+
+"""
+
+import logging
+from argparse import ArgumentParser
+from pyfilesync import get_version
+
+
+def __option_args(argv=None):
+    """
+    Parsing argument for command line program.
+
+    @param argv: Option parameters
+    @type argv: list
+    @return: parsed arguments
+    """
+    parser = ArgumentParser(description="Make backup of your files.",
+                            prog='filesync')
+    parser.add_argument("-v", "--version", action='version',
+                        version='%(prog)s {}'.format(get_version()))
+    parser.add_argument("-l", "--log", dest="logLevel",
+                        choices=['DEBUG', 'INFO', 'WARNING',
+                                 'ERROR', 'CRITICAL'],
+                        help="Set the logging level")
+    try:
+        argcomplete.autocomplete(parser)
+    except NameError:
+        pass
+    return parser.parse_args(argv)
+
+
+def main(argv=None):
+    """
+    Main function for command line program.
+
+    @param argv: Option parameters
+    @type argv: list
+    """
+    options = __option_args(argv)
```

Comment:
Parsowanie argumentów konsolowych programu

Reply:
```
optional arguments:
  -h, --help            show this help message and exit
  -v, --version         show program's version number and exit
  -l {DEBUG,INFO,WARNING,ERROR,CRITICAL}, --log {DEBUG,INFO,WARNING,ERROR,CRITICAL}
                        Set the logging level
```

To są argumenty. A może by nie powtarzać 2 razy tego samego? 
Zrobić coś w stylu
optional arguments:
```
  -h, --help            show this help message and exit
  -v, --version         show program's version number and exit
  -l, --log {DEBUG,INFO,WARNING,ERROR,CRITICAL}
                        Set the logging level
```

Reply:
Jak dla mnie to w ogole może być w ogólnym helpie bez argumentów. Można by żucić możliwe argumenty tak jak są. Już bezpośrednio w logu.

```
 usage: filesync [-h] [-v] [-l {DEBUG,INFO,WARNING,ERROR,CRITICAL}]
filesync: error: argument -l/--log: expected one argument
```

Reply:
Akurat ten opis tworzy się automatycznie, więc nie wiem czy jest sens go modyfikować jeszcze bardziej. W tym pierwszym przypadku to on sam zrobił taki opis, więc tego za bardzo nie poprawisz.